File: 71. Simplify Path/71. Simplify Path.rb
# @param {String} path
# @return {String}
def simplify_path(path)
    stack = []
    parts = path.split('/')
    
    parts.each do |part|
        next if part == '.' || part.empty?
        if part == '..'
            stack.pop if !stack.empty?
        else
            stack.push(part)
        end
    end
    
    '/' + stack.join('/')
end

File: 71. Simplify Path/71. Simplify Path.ts
function simplifyPath(path: string): string {
    const stack: string[] = [];
    const parts = path.split('/');
    
    for (const part of parts) {
        if (part === '.' || part === '') continue;
        if (part === '..') {
            if (stack.length > 0) stack.pop();
        } else {
            stack.push(part);
        }
    }
    
    return '/' + stack.join('/');
};